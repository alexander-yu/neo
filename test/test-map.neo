int double(int n) {
    return n*2;
}

float triple(float n) {
    return n*3.0;
}

void main() {
    create matrix<int> m1 = [[1,1],[2,3]];
    create matrix<float> m2 = [[1.1,2.6,3.2],[4.9,0.7,5.0]];
    println(imap(double, m1));
    println(fmap(triple, m2));
}
