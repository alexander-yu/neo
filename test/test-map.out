[[2, 2],
 [4, 6]]
[[3.3, 7.8, 9.6],
 [14.7, 2.1, 15]]
